File: Backend/Core/Features/Usuario/UsuarioEntity.cs
using Core.Features.Base;
using Core.Features.Genero;
using Core.Features.Solicitud;
using Core.Features.TipoUsuario;
using Core.Features.UsuarioTipoSolicitud;
using Core.Features.UsuarioZona;

namespace Core.Features.Usuario
{
    public class UsuarioEntity : BaseEntity
    {
        public string Nombre { get; set; }
        public string Apellido { get; set; }
        public string Codigo { get; set; }
        public string Password { get; set; }
        public string? ImagenUrl { get; set; }
        public int GeneroId { get; set; }
        public int TipoUsuarioId { get; set; }

        public IEnumerable<SolicitudEntity> Solicitudes { get; set; }
        public GeneroEntity Genero { get; set; }
        public TipoUsuarioEntity TipoUsuario { get; set; }
        public IEnumerable<UsuarioZonaEntity> UsuariosZona { get; set; }
        public IEnumerable<UsuarioTipoSolicitudEntity> UsuariosTipoSolicitud { get; set; }
    }
}

File: movilapp/lib/modules/usuario/database/usuario_entity.dart
import 'package:floor/floor.dart';

@Entity(tableName: "Usuario")
class UsuarioEntity {
  @PrimaryKey(autoGenerate: true)
  final int? id;
  final String nombre;
  final String apellido;
  final String codigo;
  final String genero;
  final String tipoUsuario;
  String? urlImagen;

  UsuarioEntity({
    required this.apellido,
    required this.codigo,
    required this.genero,
    required this.nombre,
    required this.tipoUsuario,
    this.id,
    this.urlImagen,
  });

  UsuarioEntity copyWith(
          {int? id,
          String? nombre,
          String? codigo,
          String? genero,
          String? tipoUsuario,
          String? apellido,
          String? urlImagen}) =>
      UsuarioEntity(
        id: id ?? this.id,
        apellido: apellido ?? this.apellido,
        codigo: codigo ?? this.codigo,
        genero: genero ?? this.genero,
        nombre: nombre ?? this.nombre,
        tipoUsuario: tipoUsuario ?? this.tipoUsuario,
        urlImagen: urlImagen ?? this.urlImagen,
      );
}
